mistralrs-core/src/sequence.rs

```diff
@@ -244,7 +267,14 @@ impl Sequence {
         } else if self.tokens.len().saturating_sub(self.prompt_len) == max_model_len {
             Some(StopReason::ModelLength(max_model_len))
         } else {
-            None
+            let text_acc = self.generated_text(tokenizer);
+            if self.stop_strings.iter().any(|s| text_acc.contains(s)) {
+                // TODO: a different stop reason?
+                // TODO: strip the stop string from the generated text?
```

Comment:
In general engines remove the stop token from text... But whatever, the downstream consumer can do it too easily.

Reply:
I'll create another issue for this.